plugin/pkg/managedseed/validator/admission.go

```diff
@@ -0,0 +1,373 @@
+	}
+
+	switch {
+	case managedSeed.Spec.SeedTemplate != nil:
+		// Admit seed spec against shoot
+		return v.admitSeedSpec(ctx, &managedSeed.Spec.SeedTemplate.Spec, shoot)
+
+	case managedSeed.Spec.Gardenlet != nil && managedSeed.Spec.Gardenlet.Config != nil:
+		// Decode gardenlet config from raw extension
+		gardenletConfig, err := helper.DecodeGardenletConfig(managedSeed.Spec.Gardenlet.Config)
+		if err != nil {
+			return apierrors.NewBadRequest(fmt.Sprintf("could not decode gardenlet config: %v", err))
+		}
+
+		if gardenletConfig.SeedConfig != nil {
+			// Convert seed config to internal version
+			seed, err := helper.ConvertSeed(&gardenletConfig.SeedConfig.Seed)
+			if err != nil {
+				return apierrors.NewBadRequest(fmt.Sprintf("could not convert seed config: %v", err))
+			}
+
+			// Admit seed spec against shoot
+			if err := v.admitSeedSpec(ctx, &seed.Spec, shoot); err != nil {
+				return err
+			}
+
+			// Set seed config back to gardenletConfig.SeedConfig.Seed
+			seedx, err := helper.ConvertSeedExternal(seed)
+			if err != nil {
+				return apierrors.NewBadRequest(fmt.Sprintf("could not convert seed config: %v", err))
+			}
+			gardenletConfig.SeedConfig.Seed = *seedx
+		}
+
+		// Marshal gardenlet config to JSON
+		gardenletConfigJSON, err := json.Marshal(gardenletConfig)
+		if err != nil {
+			return apierrors.NewInternalError(fmt.Errorf("could not marshal gardenlet config to JSON: %v", err))
+		}
+
+		// Set gardenlet config back to managedSeed.Spec.Gardenlet.Config
+		managedSeed.Spec.Gardenlet.Config = &runtime.RawExtension{Raw: gardenletConfigJSON}
+
+	}
+
+	return nil
+}
+
+func (v *ValidateManagedSeed) admitSeedSpec(ctx context.Context, spec *gardencore.SeedSpec, shoot *gardencore.Shoot) error {
+	// Initialize backup provider
+	if spec.Backup != nil && spec.Backup.Provider == "" {
+		spec.Backup.Provider = shoot.Spec.Provider.Type
+	}
+
+	// Initialize and validate DNS and ingress
+	ingressDomain := fmt.Sprintf("%s.%s", common.IngressPrefix, *(shoot.Spec.DNS.Domain))
+	if spec.Ingress != nil {
+		if shoot.Spec.Addons != nil && shoot.Spec.Addons.NginxIngress != nil && shoot.Spec.Addons.NginxIngress.Enabled {
+			return apierrors.NewBadRequest(fmt.Sprintf("seed ingress controller cannot be enabled on shoot %s", kutil.ObjectName(shoot)))
+		}
+
+		if spec.DNS.Provider == nil {
+			dnsProvider, err := v.getSeedDNSProvider(ctx, shoot)
+			if err != nil {
+				return err
+			}
+			spec.DNS.Provider = dnsProvider
+		}
+		// TODO If spec.DNS.Provider is not nil should we check if the user specified exactly the DNS provider we found above?
+		// I assume not, since the shoot may have multiple DNS providers
+
+		if spec.Ingress.Domain == "" {
+			spec.Ingress.Domain = ingressDomain
+		} else if spec.Ingress.Domain != ingressDomain {
+			return apierrors.NewBadRequest(fmt.Sprintf("seed ingress domain must match shoot DNS domain %s", ingressDomain))
```

Comment:
I don't think, that this is an invalid case to specify an ingress domain that doesn't match the pattern of `ingress.` + shoot domain

Reply:
OK, I wasn't sure about it. Is there something we should validate (e.g. that it contains the shoot ingress domain), or just allow the user to specify any value.

Reply:
Hmm, TBH I don't have an real overview over the shoot dns validation and also don't know the constraints for the domain used in combination with the new ingress controller. 
Maybe @BeckerMax can answer this on the new PR.
Resolving this for now and will bring it up again.